SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialVirtualWorkerTranslator.cpp

```diff
@@ -191,7 +189,7 @@ void SpatialVirtualWorkerTranslator::SendVirtualWorkerMappingUpdate()
 
 void SpatialVirtualWorkerTranslator::QueryForWorkerEntities()
 {
-	if (!bIsReady || NetDriver == nullptr)
+	if (NetDriver == nullptr)
```

Comment:
Same as above, is this ever okay to safely return? Should we check(NetDriver) instead?

Reply:
Unfortunately, right now we can't mock the NetDriver in tests, so for now we pass in nullptr in our testing. Once we untangle things enough to be able to have a mock/fake NetDriver, we can check.

Reply:
For now we could also add an Error message here. And in tests we can specify warnings/errors to supress.

Reply:
I'd rather keep tests correct as possible and not change them to hide problems in the implementation.

Reply:
I'm against having the implementation code imply returning without a null pointer is okay / expected, I think it's misleading and makes comprehending the code harder, especially since we do `check(...)`s in so many other places. 

I imagine we can't mock the NetDriver since it's such a monolith? In this case, if we were using the elements the NetDriver is required for we'd instead need to mock the static component view (for an auth check), a connection (to send a query), and the spatial receive (to register a delegate)? Assuming we haven't mocked them yet / doing a re-architecture is quite reasonably out of scope for this PR, can we settle for putting a TODO or at least a comment explicitly stating the preconditions?

Reply:
added comments.